torchvision/transforms/v2/functional/_geometry.py

```diff
@@ -188,6 +188,21 @@ def resize(
     return kernel(inpt, size=size, interpolation=interpolation, max_size=max_size, antialias=antialias)
 
 
+# This is an internal helper method for resize_image. We should put it here instead of keeping it
+# inside resize_image due to torchscript.
+# uint8 dtype support for bilinear and bicubic is limited to cpu and
+# according to our benchmarks, non-AVX CPUs should still prefer u8->f32->interpolate->u8 path for bilinear
+# For torch.compile we use uint8 input and let decomposition work
```

Comment:
I think we should be clear that the reason we always use uint8 for dynamo is simply that it doesn't support `get_cpu_capability()`, so with the suggested comment below, this comment is probably unnecessary

```suggestion
```

Reply:
A decomposition can also not support uint8 dtype, so the fact that we return True instead of False is that we believe that decomposition can work with uint8 dtype.
Even if dynamo "supported" `get_cpu_capability()` this heuristic to perform u8->f32->interpolate->u8 on non-AVX systems can be wrong for compiled version.

Reply:
OK, that makes sense. I added a suggestion above to clarify that the benchmarks were only relevant for eager.

We can merge now an iterate a bit later, but do you think our conditions could be a bit simplified? I think we should be able to do something like

```
def _do_native_uint8_resize_on_cpu(interpolation: InterpolationMode) -> bool:
	if torch._dynamo.is_compiling():
	    return True  # both bilinear and bicubic are OK, right?
	# then conditions as before
``` 

And IDK if that's true but perhaps torch.compile works for bilinear and bicubic on GPU as well, in which case we can probably write that condition much earlier?

Reply:
```
	if torch._dynamo.is_compiling():
	    return True  # both bilinear and bicubic are OK, right?
```
Well, right now, it may be safer to set `return False` due to https://github.com/pytorch/pytorch/pull/104182